puppet-checks/src/main/java/com/iadams/sonarqube/puppet/checks/FileModeCheck.java

```diff
@@ -60,22 +63,36 @@ public void init() {
 
   @Override
   public void visitNode(AstNode node) {
-    if(node.getTokenValue().equals("file")) {
+    if ("file".equals(node.getTokenValue())) {
       for (AstNode body : node.getDescendants(PuppetGrammar.RESOURCE_INST)) {
```

Comment:
@iwarapter You should be very very careful while using `getDescendants` especially on nested structures. Would it be possible to replace them by `getChildren` here?